test/k8s/services.go

```diff
@@ -584,6 +586,7 @@ Secondary Interface %s :: IPv4: (%s, %s), IPv6: (%s, %s)`,
 				"loadBalancer.mode":         "hybrid",
 				"loadBalancer.algorithm":    "random",
 				"routingMode":               "native",
+				"tunnelProtocol":            "geneve",
```

Comment:
Does that mean we're no longer auto-selecting `tunnelProtocol: geneve` when using DSR-GENEVE in native-routing? If it's possible without too much hassle, I think it would be good to preserve existing behaviour for now (to avoid upgrade regressions).

Reply:
> Does that mean we're no longer auto-selecting tunnelProtocol: geneve when using DSR-GENEVE in native-routing?

Yes. I'd personally prefer to keep this explicit, to avoid mutating the tunnel protocol configured by the users unexpectedly. Allowing the protocol to be overridden by a given module would also open up for additional complexity and possible conflicts (as different features may request different protocols) to be resolved.

But I agree that this is a possible upgrade regression. My idea was to call it out in the upgrades notes (I haven't pushed it yet though). Do you think that would be sufficient? (If users overlook that, then the agents will fail at startup with an error mentioning the protocol incompatibility). Another possibility could be adding such defaulting at the helm level, maybe setting the protocol to geneve if left to the default value, and DSR-GENEVE is enabled.

Reply:
> But I agree that this is a possible upgrade regression. My idea was to call it out in the upgrades notes (I haven't pushed it yet though).

Realistically yes, that's ok. There shouldn't be many users around, so let's not cause too much extra work for a corner-case. (adding `upgrade-impact` just in case)

There is a more general point here - imho Cilium should auto-select such config options whenever possible, and only complain on conflicts. Maybe it's ok to just have this on a ~~high~~ helm level. But we won't solve that today :slightly_smiling_face:.